staging/src/k8s.io/kubectl/pkg/explain/v2/explain.go

```diff
@@ -55,6 +56,7 @@ func printModelDescriptionWithGenerator(
 	client openapi.Client,
 	gvr schema.GroupVersionResource,
 	recursive bool,
+	verboseRecursive bool,
```

Comment:
You are completely right about this, TBH at first i wasn't sure about it too, asked about it in the original issue and didn't get any answer so i decide to put some thing here to discuss about it.

As you mention these two flags can not be used with each other, i think changing the type of `recursive` from `bool` to `string` and accepting things like `short`, `detailed` would make better sense to me. 
Like 
`kubectl explain pods.spec.containers.startupProbe --recursive=detailed` 
or 
`kubectl explain pods.spec.containers.startupProbe --recursive=short`

But it will change the usage of `--recursive=true` and i'm not sure how we deal with that. If it's ok i can implement that.
But also if we want to use depth here too idk how to catch it with one flag? maybe using something like this 
`--recursive=detailed=2` the same approach the feature gate is using 
`--feature-gates=...,GracefulNodeShutdown=true` but i'm not sure if end user will be happy about this or not.